docs/building-your-quest/writing-quest-texts/message-components.md

```diff
@@ -164,10 +164,76 @@ trigger:
 ```
 
 
+
+### Multi Select Form
+
+Allows you to ask the user for a secret (i.e API key, connection string) that will be send in an encrypted manner and won't be shared with third parties.
+This form is handled with it's own trigger `secret_submitted`
+
+```yml
+  - components:
+      - type: form
+        form:
+          id: generic_mongo_connection_string_form
+          type: secret_input_form
+```
+
+{: .note-title }
+> How will it look in Snack?
+>
+> ![secret_input_form]
+
+The submitted value can later be used to validate it's correctnes, and to be saved on the user for future use.
+The value can be found in the trigger payload, using the following notation: `"${secretValue}"`
+
+For example, here is a connection string submitted by the user:
```

Comment:
let's clean some things up and focus only on the `secret_submitted`, so keep it simple- trigger, and simple condition and then just message